vuu/src/main/scala/org/finos/vuu/core/row/InMemMapRowBuilder.scala

```diff
@@ -0,0 +1,45 @@
+package org.finos.vuu.core.row
+import org.finos.vuu.core.table.{Column, RowData, RowWithData}
+
+import scala.collection.mutable
+
+class InMemMapRowBuilder extends RowBuilder {
+
+  private val mutableMap = new mutable.HashMap[String, Any]()
+  private var key: String = ""
+  override def setLong(column: Column, v: Long): RowBuilder = {
+    mutableMap.put(column.name, v)
+    this
+  }
+
+  override def setDouble(column: Column, v: Double): RowBuilder = {
+    mutableMap.put(column.name, v)
+    this
+  }
+
+  override def setInt(column: Column, v: Int): RowBuilder = {
+    mutableMap.put(column.name, v)
+    this
+  }
+
+  override def setString(column: Column, v: String): RowBuilder = {
+    mutableMap.put(column.name, v)
+    this
+  }
+
+  override def setBoolean(column: Column, v: Boolean): RowBuilder = {
+    mutableMap.put(column.name, v)
+    this
+  }
+  override def setKey(key: String): RowBuilder = {
+    this.key = key
+    this
```

Comment:
Not sure if it applies here but would be worthwhile to think about:

It uses the same instance of the row builder so if somewhere we instantiate a row builder and share the same builder with other functions there is a chance that we might mutate the data in multiple places resulting in unexpected data changes and hard to find bugs.


For example:
```
def main() = {
  ...
  val rowBuilder = InMemMapRowBuilder().setKey("key")
  function1(rowBuilder)
  function2(rowBuilder)
}

def function1(rowBuilder: RowBuilder, ...) = {
  ...
  rowBuilder.setLong(col1, 50).asRow()
  ...
}

def function2(rowBuilder: RowBuilder) = {
  ...
  // key will be "" here, and col1 will have a value of 50
  rowBuilder.setString(col2, "ABC").asRow()
  ...
}
```

Reply:
Yes, that is very true. Each caller should getetter own I think, but I think it's fair to assume different call sites shouldn't pass them across threads.
Will update and add notthreadsafe annotation